ui/pages/confirmations/utils/confirm.ts

```diff
@@ -119,3 +122,39 @@ export const toPunycodeURL = (urlString: string): string | undefined => {
 export const stripProtocol = (urlString: string): string => {
   return urlString.replace(/^\w+:\/\//u, '');
 };
+
+const validSwapBatchTransactionCommands = [
+  '0x3593564c',
+  '0x87517c45',
+  '0x095ea7b3',
+];
+
+/**
+ * Checks if a transaction is a valid single or batch transaction
+ *
+ * @param nestedTransactions - transaction nested in a confirmation
+ */
+export function checkValidSingleOrBatchTransaction(
+  nestedTransactions?: NestedTransactionMetadata[],
+) {
+  if (!nestedTransactions || nestedTransactions?.length === 0) {
+    return;
+  }
+  if (nestedTransactions.length > 3) {
+    throw new Error(
+      'Invalid batch transaction: maximum 3 nested transactions allowed',
+    );
+  }
+  const invalidNestedTransactions = nestedTransactions.filter(
+    ({ data }) =>
+      !data ||
+      !validSwapBatchTransactionCommands.some((command) =>
+        data?.startsWith(command),
+      ),
```

Comment:
We probably need to do more than this, and check if the approvals are relative to the input token